spec/controllers/omniauth_callbacks_controller_spec.rb

```diff
@@ -0,0 +1,88 @@
+require 'rails_helper'
+
+RSpec.describe OmniauthCallbacksController, :type => :controller do
+
+  describe "facebook user persisted" do
+    before do
+      OmniAuth.config.mock_auth[:facebook] = OmniAuth::AuthHash.new({
+        :provider => 'facebook',
+        :uid => '123545',
+        :info => { :email => 'example@gmail.com' }
+        })
+
+      request.env["devise.mapping"] = Devise.mappings[:user]
+      request.env["omniauth.auth"] = OmniAuth.config.mock_auth[:facebook]
+      get :facebook
+    end
+
+    it "should successfully authenticate from Facebook" do
+      user = User.first
```

Comment:
Instead of relying on `User.first`, which seems really dependent on a particular DB state, I would use `User.find_by(email: "example@gmail.com")` and just make sure that it's not returning `nil`. For similar reasons of robustness, I would also use different uid and email values in each of the mock auth hashes.